### DISCUSSION TITLE
What's the equivalent of WebApp.Start(Owin) in .net  7.0 ?

### DISCUSSION THREAD

**Discussion by Anissa-Rouzzi** [community/community#77197] [Other Feature Feedback, Questions, & Ideas]
Hi, 

we're migrating our windows service from .net framework to .net 7.0 (core). 

we're using self-hosting via Owin (webApp.Start) with custom configuration (we add some middlewares....) but obviously, we have to replace that, 

` ....
  WebApp.Start<Startup>(url)
  ....`

so, what's the best equivalent of this in .net 7.0 ?

thanks in advance

**Comment by user667794266357** [comment]
Fghbnj
